pkg/api/types.go

```diff
@@ -466,6 +470,30 @@ type GlusterfsVolumeSource struct {
 	ReadOnly bool `json:"readOnly,omitempty"`
 }
 
+// RBDVolumeSource represents a Rados Block Device Mount that lasts the lifetime of a pod
+type RBDVolumeSource struct {
+	// Required: EndpointsName is the endpoint name that represents a collection of Ceph monitors
+	EndpointsName string `json:"endpoints"`
+	// Required: RBDImage is the rados image name
+	RBDImage string `json:"image"`
+	// Required: Filesystem type to mount.
+	// Must be a filesystem type supported by the host operating system.
+	// Ex. "ext4", "xfs", "ntfs"
+	// TODO: how do we prevent errors in the filesystem from compromising the machine
+	FSType string `json:"fsType,omitempty"`
+	// Optional: RBDPool is the rados pool name,default is rbd
+	RBDPool string `json:"pool"`
+	// Optional: RadosUser is the rados user name, default is admin
+	RadosUser string `json:"user"`
+	// Optional: Keyring is the path to key ring for RBDUser, default is /etc/ceph/keyring
+	Keyring string `json:"keyring"`
+	// Optional: Secret is the authentication secret for RBDUser, default is empty.
+	Secret string `json:"secret" description:"secret is authentication secret. If provided, it overrides keyring"`
```

Comment:
Do we really need to extract secrets right now out of volume source?  It's nice, but I don't think it gates this pull request, and has other API change implications.  Not everything needs a secret - what does it mean to apply a secret to a type that doesn't need it?

I would prefer we focus on the minimal change here.

> On Apr 21, 2015, at 4:13 PM, Paul Morie <notifications@github.com> wrote:
> 
> In pkg/api/types.go:
> 
> > +	EndpointsName string `json:"endpoints"`
> > +	// Required: RBDImage is the rados image name
> > +	RBDImage string `json:"image"`
> > +	// Required: Filesystem type to mount.
> > +	// Must be a filesystem type supported by the host operating system.
> > +	// Ex. "ext4", "xfs", "ntfs"
> > +	// TODO: how do we prevent errors in the filesystem from compromising the machine
> > +	FSType string `json:"fsType,omitempty"`
> > +	// Optional: RBDPool is the rados pool name,default is rbd
> > +	RBDPool string `json:"pool"`
> > +	// Optional: RadosUser is the rados user name, default is admin
> > +	RadosUser string `json:"user"`
> > +	// Optional: Keyring is the path to key ring for RBDUser, default is /etc/ceph/keyring
> > +	Keyring string `json:"keyring"`
> > +	// Optional: Secret is the authentication secret for RBDUser, default is empty.
> > +	Secret string `json:"secret" description:"secret is authentication secret. If provided, it overrides keyring"`
> @markturansky cool, i will rebase the kubelet-in-container PR onto yours. I guess one thing we might conceivably get into with this is if you had a plugin which wrapped another and they both need different secrets, but I don't think this exists today - we can cross that bridge when we come to it.
> …
> —
> Reply to this email directly or view it on GitHub.
>